lib/markdown-table-formatter.coffee

```diff
@@ -112,10 +112,12 @@ class TableFormatter
         widths.push(2) for c in [0..columns-1]
 
         max = (x, y) -> if x > y then x else y
+        
+        halfWidthLength = (str) -> if str.match(/[^ -~]/g) is null then str.length else str.length + str.match(/[^ -~]/g).length
```

Comment:
I'm not really happy with the fact that str.match is run twice. while it could be run once. Shouldn't it be more like

```coffeescript
halfWidthLength = (str) ->
    str.length + (str.match(/[^ -~]/g)?.length or 0)
```
?